### DISCUSSION TITLE
c pointer array

### DISCUSSION THREAD

**Discussion by zidudu** [community/community#118829] [Programming Help] [answered]
### Select Topic Area

Question

### Body

`#include <stdio.h>
int main() {
	

	int i;
	char *ptrArray[4] = { {"Korea"},{"Seoul"},{"Mapo"},{"Gimpo"} };
	for (i = 0; i < 4; i++) {
		printf("\n %s", ptrArray[i]);

	}
	ptrArray[2] = "Jongno";
	printf("\n\n");
	for (i = 0; i < 4; i++)
		printf("\n %s", ptrArray[i]);

	getchar();

}
`
this code is not compiled. why?

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @zidudu!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Feed category is solely related to conversations around the GitHub product Feed. This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!

**Comment by fdistasio** [comment]
There is a syntax error when you initialize the string array.

Array elements don't need {}, so the right initialization is:
char *ptrArray[4] = {"Korea","Seoul","Mapo","Gimpo"};

**Comment by iqnite** [comment]
Also, why is there a * before ptrArray[4]? In C, Arrays already are pointers, so you should remove it.

**Comment by MindOfBear** [comment, marked as answer]
Hello, @zidudu!

The code you provided doesn't compile because you use your ptrArray incorrectly.
Instead of using `char *ptrArray[4] = { {"Korea"},{"Seoul"},{"Mapo"},{"Gimpo"} };`
You should put elements inside without curly braces, like: `char *ptrArray[4] = {"Korea", "Seoul", "Mapo", "Gimpo"};`

Here's your full code corected

```
#include <stdio.h>
int main() {
    int i;
    char *ptrArray[4] = {"Korea", "Seoul", "Mapo", "Gimpo"};
    for (i = 0; i < 4; i++) {
        printf("\n %s", ptrArray[i]);
    }
    ptrArray[2] = "Jongno";
    printf("\n\n");
    for (i = 0; i < 4; i++) {
        printf("\n %s", ptrArray[i]);
    }
    getchar();
}
```

Have a great day!

**Reply by ebndev** [reply]
Hey @MindOfBear! Just wanted to say thanks for all of your contributions lately and for answering questions in the Community 😁

**Reply by MindOfBear** [reply]
With pleasure! I'm glad I helped you!
